controllers/drcluster_controller.go

```diff
@@ -1304,18 +1306,41 @@ func generateNF(targetCluster *ramen.DRCluster) (csiaddonsv1alpha1.NetworkFence,
 		TypeMeta:   metav1.TypeMeta{Kind: "NetworkFence", APIVersion: "csiaddons.openshift.io/v1alpha1"},
 		ObjectMeta: metav1.ObjectMeta{Name: resourceName},
 		Spec: csiaddonsv1alpha1.NetworkFenceSpec{
-			FenceState: csiaddonsv1alpha1.FenceState(targetCluster.Spec.ClusterFence),
+			FenceState: fenceState,
 			Cidrs:      targetCluster.Spec.CIDRs,
 		},
 	}
-
 	if err := fillStorageDetails(targetCluster, &nf); err != nil {
 		return nf, fmt.Errorf("failed to create network fence resource with storage detai: %w", err)
 	}
 
+	fillFenceTokenParameters(targetCluster, peerCluster, fenceState, &nf)
+
 	return nf, nil
 }
 
+func fillFenceTokenParameters(targetCluster *ramen.DRCluster, peerCluster *ramen.DRCluster, fence csiaddonsv1alpha1.FenceState, nf *csiaddonsv1alpha1.NetworkFence) {
```

Comment:
@ArbelNathan Can you please explain why you need information from the peer cluster as well for this operation?

May be explain how the tokens work in general for fencing. Also update the commit message with similar information.

Reply:
As @ArbelNathan has explained in the presentation he made to the Ramen team before this PR, we need this information to allow access to the objects from the new cluster. This is due to the fact that the only type of fencing that FS can achieve is ownership based fencing, which implies knowing the new ownership that we wish to transition to. We add that as the `unfenceToken` parameter, and that allows our driver to make the transition of objects from under the `fenceToken` ownership to the `unfenceToken` ownership.

We appreciate that it's a deviation from how NetworkFence object is designed to work, but given the limited time, we tried to do the minimum to make it work with whatever was already available. This is what we believe we got approval for in the corresponding integration slack thread.

We will probably need to update this code a little and add explanatory comments, since formally, the choice of the fence and unfence tokens must be based on the roles of the clusters we are about to generate the NetworkFence objects on, rather than the fencing state of the object.

```
                    Hub
                   =====

    [DRClusterA]:             [DRClusterB]:
    fence-token               fence-token
  /             \           /             \
 |               \         /               \
 |                \       /                 |
 |                 \     /                  |
 |  Cluster A       \   /     Cluster B     |
 |  =========        \ /      =========     |
 |                    /                     |
 |  [NF]:            / \      [NF]:         |
  \ fenceToken  ----     ---- fenceToken   /
    unfenceToken              unfenceToken
```

Reply:
Thanks for the explanation! Please add the comments and explanation in a follow up PR.